pkgs/os-specific/linux/open-iscsi/default.nix

```
@@ -1,36 +1,38 @@
-{ stdenv, fetchFromGitHub, automake, autoconf, libtool, gettext, utillinux, openisns, openssl, kmod }:
+{ stdenv, fetchFromGitHub, automake, autoconf, libtool, gettext,
+utillinux, openisns, openssl, kmod, perl, systemd, pkgconf }:
+
 stdenv.mkDerivation rec {
   name = "open-iscsi-${version}";
-  version = "2.0-873-${stdenv.lib.substring 0 7 src.rev}";
+  version = "2.0-877";
+
+  buildInputs = [ automake autoconf libtool gettext utillinux openisns.lib openssl kmod perl systemd pkgconf ];
 
-  buildInputs = [ automake autoconf libtool gettext utillinux openisns.lib openssl kmod ];
-  
   src = fetchFromGitHub {
     owner = "open-iscsi";
     repo = "open-iscsi";
-    rev = "4c1f2d90ef1c73e33d9f1e4ae9c206ffe015a8f9";
-    sha256 = "0h030zk4zih3l8z5662b3kcifdxlakbwwkz1afb7yf0cicds7va8";
+    rev = "refs/tags/${version}";
```

With the above change that should work now:
```suggestion
    rev = "${version}";
```

---

Or just `rev = version;` would be shorter but it is not sufficient here since the tag is `2.0.877` and not `2.0-877`.

Cleaner to use only dots in `version` to solve this

---

I changed it to use only dots, but mind that the tool itself reports 2.0-877 as the version.
```
$ result/bin/iscsistart -v
iscsistart version 2.0-877
```

---

I would recommend the dotted variant. That is how they tag the versions on github. With the dots the auto updater will be able to pick up the next updates.